src/System.IO.Compression/src/System/IO/Compression/DeflateManaged/InflaterManaged.cs

```diff
@@ -64,17 +64,20 @@ internal sealed class InflaterManaged
         private readonly byte[] _codeLengthTreeCodeLength;
         private readonly bool _deflate64;
         private HuffmanTree _codeLengthTree;
+        private long _uncompressedSize = -1;
```

Comment:
This field should be readonly and not initialized, since it's only unconditionally assigned in the solitary constructor
```suggestion
        private readonly long _uncompressedSize;
```

Reply:
if we will not initialize it here it will have value of 0 by default, uncompressedSize set from contractor also can have value of 0 in case empty file, i.e. 0 is valid uncompressed size and we initializing it here to -1 to make sure it is not set with the constructor so that we will not account it in further calculation/inflation

Reply:
Is the concern that if we add some other constructor in the future they might forget to set this value? That seems like an odd thing to guard against, because as the code currently stands, this -1 doesn't perform any function other than to confuse the reader, like it confused me.  It _looks_ like the normal behavior is to do the indeterministic reading, but there is actually no code path where that is true.

Reply:
I agree with @ChadNedzlek ... there isn't (that I see) a second constructor in this type.  Since the one constructor unconditionally assigns this value, this initializer has no value.  (On other types it might make sense to have an initializer for the value)

Reply:
yes there is no other constractor initializing it, the point is to not broke existing apps using this API.  Even this Inflators are internal DeflateStreams using them are public API, for now this fix is intended for internal deflate streams which are using the only constructor setting this value. I can make it readonly and add defaults (still will be -1 as 0 is valid size and should be counted) to the all other the constractors if that is what you are suggesting

Reply:
The field should be readonly because it is only updated in the constructor, and that's good (and readonly enforces this).

It should not have an initializer value because it's **always** assigned/overwritten in the constructor.  If you made the initializer value here be 1048576 it wouldn't change anything, because the constructor is unconditionally replacing it.

Reply:
oh I confused it with DeflateStream constructors where i was limiting output initially, just saw that there is no other constructors for Inflators, so yes you guys right it not need to be initialized outside of constructor, will update